db/migrate/20160922144222_change_fact_name_to_puppet_fact_name.rb

```diff
@@ -0,0 +1,13 @@
+class ChangeFactNameToPuppetFactName < ActiveRecord::Migration
+  def up
+    FactName.select { |f| f.type == 'FactName' }.each do |fact|
```

Comment:
This loads all fact names in memory (18328 in my case) and then only select among instantiated objects, why not using `FactName.where(:type => 'FactName')`? or even better using `FactName.update_all` to make just one query skipping validations?